server/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilder.java

```diff
@@ -101,7 +101,7 @@ public final AggregationBuilder rewrite(QueryRewriteContext context) throws IOEx
         if (rewritten == this) {
             return rewritten;
         }
-        if (getMetaData() != null && rewritten.getMetaData() == null) {
+        if (getMetaData() != null && (rewritten.getMetaData() == null || rewritten.getMetaData().isEmpty())) {
             rewritten.setMetaData(getMetaData());
         }
```

Comment:
what prevents us from always blindly setting the metadata? Rewriting should never modify it?

Reply:
good point. I'll change it to do this since it simplifies things